File: IQ/Classes/Utility/ErrorLog.vb
Public Class ErrorLog
    Private Shared ErrorList As LoggingList(Of String) = New LoggingList(Of String)()
    Public Shared Sub Add(ex As Exception)
        If ex IsNot Nothing Then
            ErrorList.Add(String.Format("{0}\r\n{1}\r\n{2}", ex.Message, If(ex.InnerException IsNot Nothing, ex.InnerException.Message, String.Empty), ex.StackTrace))
            Try
                dataAccess.da.DBExecutesql(String.Format("INSERT INTO ErrorLog (DateTime,Message,StackTrace,InnerException) VALUES ({0},{1},{2},{3})", dataAccess.da.UniversalDate(DateTime.Now), dataAccess.da.SqlEncode(ex.Message), dataAccess.da.SqlEncode(ex.StackTrace), dataAccess.da.SqlEncode(If(ex.InnerException IsNot Nothing, ex.InnerException.Message, String.Empty))))
            Catch exe As Exception
                'Dont crash, thats what we are trying to avoid!!
            End Try
        End If

    End Sub


End Class

File: IQ/Classes/Utility/ErrorLog.cs
public class ErrorLog
{
    public ErrorLog()
    {
        // VBConversions Note: Non-static class variable initialization is below.  Class variables cannot be initially assigned non-static values in C#.
        ErrorList = new LoggingList<string>();

    }
    private static LoggingList<string> ErrorList; // VBConversions Note: Initial value cannot be assigned here since it is non-static.  Assignment has been moved to the class constructors.
    public static void Add(Exception ex)
    {
        if (ex != null)
        {
            ErrorList.Add(string.Format("{0}\\r\\n{1}\\r\\n{2}", ex.Message, ex.InnerException != null ? ex.InnerException.Message : string.Empty, ex.StackTrace));
            try
            {
                dataAccess.da.DBExecutesql(string.Format("INSERT INTO ErrorLog (DateTime,Message,StackTrace,InnerException) VALUES ({0},{1},{2},{3})", dataAccess.da.UniversalDate(DateTime.Now), dataAccess.da.SqlEncode(ex.Message), dataAccess.da.SqlEncode(ex.StackTrace), dataAccess.da.SqlEncode(ex.InnerException != null ? ex.InnerException.Message : string.Empty)));
            }
            catch (Exception)
            {
                //Dont crash, thats what we are trying to avoid!!
            }
        }

    }


}